### PR TITLE
Add user fiendly description for CompiledIsNullExpression and other plan nodes 

### PR DESCRIPTION
**Pull request by amitvc**
Based on the discussions on https://github.com/diennea/herddb/issues/471 we have overriden the toString method on ValuesOp so we display a better message in explain query.

### CODE REVIEW

**Comment by eolivelli** [comment]
Can you please do the same on other nodes of the plan?
Like SortOp.....

**Comment by eolivelli** [comment]
Travis failed.
Please check

**Comment by amitvc** [comment]
Yes will do.

**Comment by amitvc** [comment]
@eolivelli will do

**Comment by eolivelli** [comment]
It looks like that every CompiledSQLExpression subclass needs a toString

if you run all of the tests with -Dherddb.planner.dumpqueryloglevel=INFO you will see all of the plans and you will see how ugly they appear now :-)

File: herddb-core/src/main/java/herddb/model/planner/SemiJoinOp.java [hunk 1]
```diff
@@ -82,6 +82,10 @@ public StatementExecutionResult execute(
         );
         EnumerableDataScanner joinedScanner = new EnumerableDataScanner(resRight.dataScanner.getTransaction(), fieldNames, columns, result, resLeft.dataScanner, resRight.dataScanner);
         return new ScanResult(resTransactionId, joinedScanner);
+    }
 
+    @Override
+    public String toString() {
+        return String.format("SemiJoinOp {leftKeySize = %d rightKeySize = %d }", leftKeys.length , rightKeys.length);
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/model/planner/SemiJoinOp.java:89]: Shall we tell more about the inputs? Like calling testing on them?

File: herddb-core/src/main/java/herddb/model/planner/LimitOp.java [hunk 2]
```diff
@@ -118,4 +118,9 @@ public PlannerOp optimize() {
         }
         return this;
     }
+
+    @Override
+    public String toString() {
+        return "LimitOp{value = " + this.maxRows.toString() + " }";
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/model/planner/LimitOp.java:124]: We should recursively call testing on the input op

File: herddb-core/src/main/java/herddb/model/planner/SemiJoinOp.java (hunk 1 above)
Comment (amitvc) [reply herddb-core/src/main/java/herddb/model/planner/SemiJoinOp.java:89]: @eolivelli Can you please explain what do you mean by testing?

File: herddb-core/src/main/java/herddb/model/planner/LimitOp.java (hunk 2 above)
Comment (amitvc) [reply herddb-core/src/main/java/herddb/model/planner/LimitOp.java:124]: Will do.

**Comment by eolivelli** [comment]
Is this patch ready for review?

**Comment by amitvc** [comment]
@eolivelli Yes.

File: herddb-core/src/main/java/herddb/model/commands/InsertStatement.java [hunk 3]
```diff
@@ -62,4 +62,9 @@ public boolean isUpsert() {
         return upsert;
     }
 
+    @Override
+    public String toString() {
+        return String.format("{InsertStatement={ keyfunction=%s valuesFunction=%s}}",
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/model/commands/InsertStatement.java:67]: remove '{}' that are wrapping InsertStatement

File: herddb-core/src/main/java/herddb/model/planner/SimpleInsertOp.java [hunk 4]
```diff
@@ -66,4 +66,8 @@ public boolean isSimpleStatementWrapper() {
         return true;
     }
 
+    @Override
+    public String toString() {
+        return String.format("{SimpleInsert= statement %s }", statement.toString());
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/model/planner/SimpleInsertOp.java:71]: Remove '{}' that are wrapping SimpleInsert

File: herddb-core/src/main/java/herddb/model/planner/UpdateOp.java [hunk 5]
```diff
@@ -173,4 +173,9 @@ public void accept(StatementExecutionResult res, Throwable error) {
         }
         return Wrapper.unwrap(this, clazz);
     }
+
+    @Override
+    public String toString() {
+        return String.format("{UpdateOp=[ input=%s recordFunction=%s}", input, recordFunction.toString());
```
Comment (eolivelli) [inline herddb-core/src/main/java/herddb/model/planner/UpdateOp.java:179]: remove '{}' that are wrapping UpdateOp

File: herddb-core/src/test/java/herddb/sql/CalcitePlannerTest.java [hunk 6]
```diff
@@ -486,6 +486,44 @@ public void explainPlanTest() throws Exception {
         }
     }
 
+    @Test
+    public void explainPlanTestIsNullExpression() throws Exception {
```
Comment (eolivelli) [inline herddb-core/src/test/java/herddb/sql/CalcitePlannerTest.java:490]: is this test useful ?

File: herddb-core/src/main/java/herddb/model/commands/InsertStatement.java (hunk 3 above)
Comment (amitvc) [reply herddb-core/src/main/java/herddb/model/commands/InsertStatement.java:67]: Done.

File: herddb-core/src/test/java/herddb/sql/CalcitePlannerTest.java (hunk 6 above)
Comment (amitvc) [reply herddb-core/src/test/java/herddb/sql/CalcitePlannerTest.java:490]: In the current state, it is not. I removed it.

**Comment by amitvc** [comment]
@eolivelli Pushed suggested changes

**Comment by eolivelli** [comment]
@amitvc 
Travis failed on checkstyle, please fix it, then we are good to go
```

[ERROR] /home/travis/build/diennea/herddb/herddb-core/src/main/java/herddb/model/planner/UnionAllOp.java:33: Extra separation in import group before 'java.util.Arrays' [ImportOrder]
Audit done.
[INFO] There is 1 error reported by Checkstyle 8.30 with checkstyle.xml ruleset.
[ERROR] src/main/java/herddb/model/planner/UnionAllOp.java:[33] (imports) ImportOrder: Extra separation in import group before 'java.util.Arrays'
```

**Comment by eolivelli** [comment]
merged to master branch
thank you @amitvc

**Pull request merged**